datafusion/physical-plan/src/sorts/sort_preserving_merge.rs

```diff
@@ -98,19 +99,56 @@ pub struct SortPreservingMergeExec {
     ///
     /// See [`Self::with_round_robin_repartition`] for more information.
     enable_round_robin_repartition: bool,
+    ///
+    progressive_eval_exec: Option<Arc<ProgressiveEvalExec>>,
 }
 
 impl SortPreservingMergeExec {
     /// Create a new sort execution plan
     pub fn new(expr: LexOrdering, input: Arc<dyn ExecutionPlan>) -> Self {
         let cache = Self::compute_properties(&input, expr.clone());
+
+        // Todo: check if partition statistic is accurate
+        // Organize the input partitions into chains,
+        // where elements of each chain are input partitions that are
+        // non-overlapping, and each chain is ordered internally by their min/max statistics.
+        let partition_groups = input
+            .statistics_by_partition()
```

Comment:
This partition group mapping (swapping ordering by statistics) is also the approach we are using.